### ISSUE TITLE
Sources published to npm

### ISSUE THREAD

**Issue by grigasp** [planttheidea/micro-memoize#97]
Is there any reason to include source files in the published package?

![image](https://user-images.githubusercontent.com/35135765/236436250-979a12be-70e8-4e86-9db1-a0f4a69eb305.png)

**Comment by planttheidea** [comment]
The reason is because the `index.d.ts` file sources these files. Admittedly not the cleanest setup, but I haven't had time nor need to circle back to clean it up until this point.

As a bit of feedback, if you are facing an issue I would provide some details as to what the issue actually is, because the question alone does not convey that there is a problem other than the academic "is this necessary?" musing. In the linked issue, it seems that some API extractor doesn't like the use of non-definition files defining types? I don't have much insight into what tools you're using so it is tough to provide guidance here.

**Comment by planttheidea** [comment]
I've published `4.1.0-beta.0` as an attempt to resolve this. If you want to try that out and tell me if it resolves the issue, I can move forward with making it a stable version.

**Comment by grigasp** [comment]
@planttheidea thanks, I confirmed that `@microsoft/api-extractor` doesn't complain about this anymore when using version `4.1.0-beta.1`.

As for more details, `api-extractor` was failing with the following error:
```
./node_modules/.pnpm/micro-memoize@4.0.11/node_modules/micro-memoize/src/types.ts:1:1 - (ae-wrong-input-file-type) Incorrect file type; API Extractor expects to analyze compiler outputs with the .d.ts file extension. Troubleshooting tips: https://api-extractor.com/link/dts-error
```

https://api-extractor.com/link/dts-error states:
> Is an NPM package dependency exporting TypeScript source files instead of .d.ts declaration files? Ask the package maintainer to follow standard practice of distributing prebuilt outputs, not source code.

In addition, delivering source files with the package unnecessarily bloats its size.

**Comment by planttheidea** [comment]
This change has been included in the `4.1.0` release. If you have any more issues, let me know!
